pwiz_tools/Skyline/Util/Util.cs

```diff
@@ -2001,6 +2040,14 @@ public static void AreEqual(double expected, double actual, double delta, string
 
         public static void Fail(string error = "")
         {
+            if (InvokeDebuggerOnFail)

```

Comment:
Let's just put this all in AssertEx for now, which makes it more limited to testing. It will likely produce cleaner code. In general, it was a mistake that you seemed to have made to use Assume in tests instead of Assert[Ex]. Maybe we should clean that up. Assume is really supposed to be for implementation code and not testing. This just further muddies that picture.

Reply:
OK, per our video chat we convinced ourselves that it does need to be in Assume as well as AssertEx if it's going to catch the bug we're chasing in TestLibraryBuild. where the Assert.Fail passes through Assume rather than AssertEx.

The ctor now just takes a simple bool, for best generality and simplicity, with a default of True,